Dart 3 migration

morebytes opened the pull request:
## Summary
Upgrades Campus Mobile repo to latest version of Flutter and Dart. 


## Changelog

- Upgrades to Flutter 3.24 and Dart 3.5 
- Upgrades Firebase packages to latest versions
- Upgrades any other necessary packages


## Test Plan
[PR-2080-Test-Plan-7.29.0-QA-iOS.xlsx](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7B9FC8CFB9-30F2-4179-9CFB-28020A983182%7D&file=[PR-2080-Test-Plan-7.29.0-QA-3283.xlsx](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7B9FC8CFB9-30F2-4179-9CFB-28020A983182%7D&file=PR-2080-Test-Plan-7.29.0-QA-3283.xlsx&action=default&mobileredirect=true)&action=default&mobileredirect=true)
[PR-2080-Test-Plan-7.29.0-QA-Android.xlsx](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7BCAA89DC5-07B5-4BDD-B657-68A8666C3670%7D&file=[PR-2080-Test-Plan-7.29.0-QA-3284.xlsx](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7BCAA89DC5-07B5-4BDD-B657-68A8666C3670%7D&file=PR-2080-Test-Plan-7.29.0-QA-3284.xlsx&action=default&mobileredirect=true)&action=default&mobileredirect=true)

morebytes committed:
Migrated to Dart 3 and upgraded all firebase and other packages that were too outdated for Dart 3.

morebytes committed:
- Upgraded one more critical package that had bugfixes for latest Flutter version
- Migrated to the very latest version of Dart 3.

morebytes committed:
- Cleaned up unused imports and code after Dart 3 migration

morebytes committed:
Fixed runZonedGuarded in main.dart
Removed commented out code in pubspec.yaml
Added proper return types to async startup functions in main.dart

morebytes committed:
Removed commented out code from pubspec

morebytes committed:
Removed some redundant null checks
Upgraded Kotlin build tools

morebytes committed:
Added temporary line to gradle file for outdated libraries (should be removed later on)
Upgraded audio packages in pubspec.yaml

morebytes committed:
Upgraded flutter_scandit package to latest version. Note: this package is discontinued and should eventually be replaced with the new official package. Will require significant rewriting.

morebytes committed:
Upgraded podfile to test PR CI/CD build

c3bryant commented on ios/Podfile:
```diff
@@ -1,5 +1,5 @@
 # Uncomment this line to define a global platform for your project
-platform :ios, '11.0'
+platform :ios, '12.0'
```
Is this a required change?

morebytes replied:
Minimum iOS version for latest Flutter is 12 [official doc](https://docs.flutter.dev/deployment/ios)

morebytes replied:
Most older iPhones run 12+ iOS

c3bryant commented on lib/ui/triton_media/triton_media_detail_view.dart:
```diff
@@ -418,7 +418,7 @@ class SeekBarState extends State<SeekBar> {
                       ?.group(1) ??
                   '$_remaining',
-              style: Theme.of(context).textTheme.caption),
+              style: Theme.of(context).textTheme.bodySmall),
```
Is this equivalent?

morebytes replied:
Yes

c3bryant commented on pubspec.yaml:
```diff
@@ -3,40 +3,34 @@ description: Mobile app platform for campus.
-      ref: scandit-6.6.3
-  flutter_linkify: 5.0.2
-  flutter_local_notifications: 9.9.1
+  flutter_scandit_plugin: ^3.0.0
```
We may want to update our fork if there is a chance this upstream will go away.

morebytes replied:
Agreed, and also update our fork specifically to target Flutter 3.24 and latest supported iOS/Android versions if possible

c3bryant commented on lib/ui/map/map_search_view.dart:
```diff
@@ -3,7 +3,7 @@ import 'package:campus_mobile_experimental/ui/common/container_view.dart';
 import 'package:campus_mobile_experimental/ui/map/search_bar.dart';
 import 'package:campus_mobile_experimental/ui/map/search_history_list.dart';
-import 'package:flutter/material.dart';
+import 'package:flutter/material.dart' hide SearchBar;
```
What is the significance of this?

morebytes replied:
Resolves a naming conflict due to Flutter breaking changes

c3bryant committed:
Remove ios googleservice info

c3bryant committed:
Fix ios build issues
Removed carets from dependencies
Bumped iOS min version to 14.0
Point flutter_scandit_plugin to experimental version
Update podfile

c3bryant committed:
Fix ios build issues part 2
Applied consistent dependency version references
Revert to flutter_scandit_plugin fork version 6.6.3

klortiz13 committed:
added foregroundcolor next to the background color to make the text white in dark mode

morebytes commented:
Thank you @klortiz13 for fixing the dark mode text bug. Now that it's out of the way, we are nearing completion.

morebytes committed:
Merge remote-tracking branch 'origin/experimental' into dart_3_migration

# Conflicts:
#	ios/Podfile.lock
#	lib/app_networking.dart
#	lib/core/providers/user.dart
#	lib/main.dart
#	lib/ui/my_chart/my_chart_card.dart
#	lib/ui/profile/cards.dart
#	lib/ui/triton_media/triton_media_card.dart
#	lib/ui/triton_media/triton_media_detail_view.dart
#	lib/ui/triton_media/triton_media_tile.dart
#	pubspec.lock
#	pubspec.yaml

morebytes committed:
Merged w/ latest experimental! TODO: regenerate podfile.lock

c3bryant committed:
Regenerate podfile.lock

morebytes committed:
Refactored several things to improve migratability

morebytes committed:
- Removed dead code and redundant imports
- Migrated discontinued packages: connectivity -> connectivity_plus, and package_info -> package_info_plus
- Upgraded provider package to get vital bugfixes, security patches, and performance improvements
- Upgraded dots_indicator and rewrote old onboarding_screen code to support more accurate page handling
- Upgraded percent_indicator, shared_preferences, flutter_sticky_header, intl, url_launcher, device_info_plus, permission_handler

morebytes committed:
- Removed seemingly dead code for unused animations. This can be reverted in the future if the animations were actually being used for something.
- Refactored some functions in onboarding_screen to be more succinct and efficient

morebytes committed:
Upgraded build_runner for potential performance improvements and to pave the way towards upgrading Hive and overall build system

klortiz13 committed:
1) Fixed back arrow in onboarding page (black to white)
2) Fixed Apple Logo in weather card (to grey)
3) Fixed classes card so that it displays properly in both themes (by commenting out indexing line at ui/classes/upcomingclasses.dart line 45) AND adding explicit colors (so it doesn't inherit from parent).
4) Commented out the wifi card (ui/home/home.dart line 245) so that the home error is gone (needs fixing or removal).
5) Blocker -> Can't load the profile page after git pulling.
6) Fixed the passwordObscured logo in profile page.

klortiz13 committed:
Changed colors for better visibility in dark mode

klortiz13 committed:
Upcoming class color change from dark blue to inherited color, Changed the color of the hidePassword logo, Fixed the cropped class by adjusting the size of the class box in the classes card to 0.34

klortiz13 committed:
fixed the hard-to-see-in-dark-mode highlighted tile in the class card

morebytes committed:
Expanded on the selectedColor solution from previous commit with a more general solution

morebytes committed:
Oops! Fixed small bug from previous commit...

morebytes reviewed (commented):
Thoroughly tested, seems to be working. Will upgrade CodeMagic build pipeline after merging this

morebytes committed:
Inverted small logic bug in Speed Test
